deltacat/compute/compactor_v2/compaction_session.py

```diff
@@ -629,11 +630,24 @@ def merge_input_provider(index, item):
         input_inflation=input_inflation,
         input_average_record_size_bytes=input_average_record_size_bytes,
     )
-
     logger.info(
         f"partition-{params.source_partition_locator.partition_values},"
         f"compacted at: {params.last_stream_position_to_compact},"
     )
+    is_inplace_compacted: bool = (
+        params.source_partition_locator == params.destination_partition_locator
+    )
+    if (
+        is_inplace_compacted
+        and compacted_partition.locator != rcf_source_partition_locator
+    ):
+        # NOTE: Concurrent compactions may affect the partitions pointed to in these equality checks
+        logger.warning(
+            "Overriding round completion file source partition locator as in-place compacted. "
+            + f"Got compacted partition partition_id of {compacted_partition.locator.partition_id} "
+            f"and rcf source partition_id of {rcf_source_partition_locator.partition_id}."
```

Comment:
We are missing unit tests for this functionality.

Reply:
Now added